### PR TITLE
[stable/fluentd] Add support for fluentd autoscaling

### LINKED ISSUE #16224: [stable/fluentd] Add support for auto scale
We use fluentd in some cases a log aggregator. As such, we would like it to be able to automatically to scale up and down.

### PR DESCRIPTION
**Pull request by sharon-vendrov**
#### What this PR does / why we need it:
This PR adds support for fluentd autoscaling.
We use fluentd as our log aggregator, in order to be able to handle log load, we need to be able to autoscale fluentd replicas.
This PR is major change, as we change the usage of (deployments & pvc) to (statefulset & hpa).

#### Which issue this PR fixes
  - fixes #16224

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by sharon-vendrov** [comment]
@rendhalver please approve tests execution

**Comment by sharon-vendrov** [comment]
@hectorj2f @Miouge1 - Please trigger tets on my PR

File: stable/fluentd/templates/deployment.yaml [hunk 1]
```diff
@@ -94,9 +98,19 @@ spec:
...
             name: {{ template "fluentd.fullname" . }}
         - name: buffer
-          {{- if .Values.persistence.enabled }}
+          {{- if and .Values.persistence.enabled (not .Values.autoscaling.enabled) }}
           persistentVolumeClaim:
             claimName: {{ template "fluentd.fullname" . }}
+          {{- else if and .Values.persistence.enabled (.Values.autoscaling.enabled) }}
+          volumeClaimTemplates:
+          - metadata:
+              name: buffer
```
Comment (hectorj2f) [inline stable/fluentd/templates/deployment.yaml:107]: name: `{{ template "fluentd.fullname" . }}`

File: stable/fluentd/values.yaml [hunk 2]
```diff
@@ -167,3 +167,22 @@ nodeSelector: {}
...
+  enabled: false
+  minReplicas: 2
+  maxReplicas: 5
+  metrics:
+    - type: Resource
+      resource:
+        name: cpu
+        target:
+          type: Utilization
+          averageUtilization: 60
```
Comment (hectorj2f) [inline stable/fluentd/values.yaml:182]: these settings are a bit opinionated or focused on a specific threshold. All these settings should be configurable as well.

File: stable/fluentd/templates/deployment.yaml (hunk 1 above)
Comment (sharon-vendrov) [reply stable/fluentd/templates/deployment.yaml:107]: "name: buffer", is the current configuration of the deployment.yml, we would like to change it?

File: stable/fluentd/values.yaml (hunk 2 above)
Comment (sharon-vendrov) [reply stable/fluentd/values.yaml:182]: These are only default, these are totally configurable.
The chart consume the entire "metrics" field  AS-IS

File: stable/fluentd/templates/deployment.yaml (hunk 1 above)
Comment (hectorj2f) [reply stable/fluentd/templates/deployment.yaml:107]: let's keep it as it is.

File: stable/fluentd/values.yaml (hunk 2 above)
Comment (hectorj2f) [reply stable/fluentd/values.yaml:182]: okay, it sounds good to me. I missed this structure somehow.

**Pull request merged**